gunicorn/systemd.py

```diff
@@ -43,3 +44,39 @@ def listen_fds(unset_environment=True):
         os.environ.pop('LISTEN_FDS', None)
 
     return fds
+
+
+def sd_notify(state, unset_environment=False, debug=False):
```

Comment:
Currently, there is no way to actually override the default values of `unset_environment` and `debug`. I think we either should find a way to customize the behavior (e.g. by passing `debug=self.cfg.reload` to `systemd.sd_notify()` in `Arbiter.start()`) or drop these arguments.

I don't use systemd, but I'm not sure hiding exceptions by default is a good idea.

Reply:
> but I'm not sure hiding exceptions by default is a good idea.

in `arbiter.py` the call to `systemd.sd_notify` is unconditional, so in the weird case that the environment variable exists but is not quite the socket expected, I've added the additional try/except. other than that, there's not much to fail (I think).

> Currently, there is no way to actually override the default values of unset_environment and debug

I've left those there in case some other uses of sd_notify appears in the future.

Reply:
i think i would remove the default from the signature if it's not needed and maybe add a comment about it. Thoughts?

Reply:
looking at the code if the unset environnement is really needed i would expose it as an option.

for the debug log I would just reuse the logger instance and pass it to the function if possible.

Reply:
maybe I should use `self.log.exception` ?

Reply:
how about d71e746c ?

Reply:
looks fine for me. is ˋunset_environment` is needed maybe it can be a setting from the config ?